extensions-contrib/kubernetes-overlord-extensions/README.md

```diff
@@ -0,0 +1,222 @@
+    subjects:
+      - kind: ServiceAccount
+        name: <druid-overlord-k8s-service-account>
+        namespace: <druid-namespace>
+    roleRef:
+      kind: Role
+      name: druid-k8s-task-scheduler
+      apiGroup: rbac.authorization.k8s.io
+    ```
+
+2. Configure the overlord's runtime properties to use the Kubernetes task scheduler.
+    ```
+    druid.indexer.runner.type=k8s
+    druid.indexer.task.encapsulatedTask=true
+    druid.indexer.runner.namespace=<druid-namespace>
+    druid.indexer.runner.capacity=100 (number of concurrent jobs that can be sent to kubernetes at once)
+    ```
+
+3. Choose a Task Adapter (determines how tasks will be converted to a Kubernetes Job)
+    #### Option 1. Copy the overlord's pod template to use as the base job template
+    This adapter takes the overlord's own pod specification and injects needed information (like the task id) on top of it.
+
+    Required overlord runtime properties 
+    ```
+    druid.indexer.runner.k8s.adapter.type=overlordSingleContainer (defaults to this if not specified)
+    druid.indexer.runner.k8s.adapter.type=overlordMultiContainer (same as overlordSingleContainer except sidecars are respected)
+    ```
+    Additional overlord runtime properties
+    ```
+    druid.indexer.runner.javaOptsArray=[] (JVM args for the peon running in the Job, If -Xmx is specified, the adapter will attempt to scale the Kubernetes requests of the pod to correspond with it)
+    druid.indexer.runner.labels={} (Additional labels to add to the Job)
+    druid.indexer.runner.annotations={} (Additional annotations to add to the Job)
+    druid.indexer.runner.peonMonitors={} (Override the list of Druid monitors for the peon, otherwise the overlord's list will be used)
+    ```
+
+    #### Option 2. Bring your own pod template to use as the base job pod template
+    This adapter expects a [PodTemplate](https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates) to be available on the overlord's file system.
+    This pod template is used as the base of the job template (additional info like the task id is injected on top of it).
+    You can override things like labels, environment variables, resources, annotation, or even the base image with this template.
+
+    Required overlord runtime properties
+    ```
+    druid.indexer.runner.k8s.adapter.type=customTemplateAdapter
+    druid.indexer.runner.k8s.podTemplate.base=/opt/druid/conf/druid/cluster/pod-template.yaml` (location of pod template on overlord)
+    ```
+
+    The following PodTemplate is an example that uses the regular druid docker image.
+    ```
+    apiVersion: "v1"
+    kind: "PodTemplate"
+    template:
+      metadata:
+        annotations:
+          custom-annotation: "hello"
+        labels:
+          custom-label: "hello"
+      spec:
+        affinity: {}
+        containers:
+        - command:
+            - sh
+            - -c
+            - |
+              /peon.sh /druid/data 1
```

Comment:
`/peon.sh`

Where does this file come from?

Reply:
this is from the druid docker image https://github.com/apache/druid/blob/master/distribution/docker/peon.sh, although if you build your own docker image the command might be different